lib/internal/errors.js

```diff
@@ -204,6 +207,17 @@ const addCodeToName = hideStackFrames(function addCodeToName(err, name, code) {
   }
 });
 
+function isErrorStackTraceLimitWritable() {
+  const desc = ObjectGetOwnPropertyDescriptor(Error, 'stackTraceLimit');
+  if (desc === undefined) {
+    return ObjectIsExtensible(Error);
+  }
+
+  return ObjectPrototypeHasOwnProperty(desc, 'writable') ?
+    desc.writable :
+    ObjectPrototypeHasOwnProperty(desc, 'set');
```

Comment:
Note that this condition will always return `true`, because `set` is always an own property for accessors, the correct check is:
```suggestion
    typeof desc.set === "function";
```

## See:
- <https://tc39.es/ecma262/#table-data-property-attributes>
- <https://tc39.es/ecma262/#table-accessor-property-attributes>
- <https://tc39.es/ecma262/#sec-frompropertydescriptor>

Reply:
I don't think that's true, `Object.getOwnPropertyDescriptor(Error, 'stackTraceLimit').hasOwnProperty('set') === false`.

Reply:
That is only true if the descriptor is a data descriptor, for accessors they do have one:

```console
> Object.defineProperty(Error, 'captureStackTrace', {get: () => {}});
[Function: Error] { stackTraceLimit: 10 }
> Object.getOwnPropertyDescriptor(Error, 'captureStackTrace').hasOwnProperty('set')
true
```

Reply:
This makes me think we need a test for this

Reply:
So this ?
```suggestion
    ObjectPrototypeHasOwnProperty(desc, 'set') && desc.set !== undefined;
```

Reply:
Oh no need to test for the 'own' part at this point, I think I got it now.
```suggestion
    desc.set !== undefined;
```